src/backend/marsha/core/api/live_session.py

```diff
@@ -299,3 +301,73 @@ def set_display_name(self, request):
                 )
 
             raise error
+
+    @action(
+        detail=False,
+        methods=["get"],
+        permission_classes=[permissions.IsTokenInstructor | permissions.IsTokenAdmin],
+        url_path="list_attendances",
+    )
+    def list_attendances(self, request, *args, **kwargs):
+        """
+        Retrieve the list of attendances computed for livesessions where is_registered
+        equals True or with the live_attendance field not empty. As we request a list
+        of statistics, we want the timeline of the video to be identical for all
+        students so we pass the list of timestamp by context.
+        The same one will then be used for all the students on this list.
+        Serializer is cached so the listing don't get recalculated too often
+        """
+
+        cache_key = (
+            f"attendances:video:{self.request.user.id}"
+            f"offset:{self.request.query_params.get('offset')}"
+            f"limit:{self.request.query_params.get('limit')}"
+        )
+        if cached_data := cache.get(cache_key, None):
+            return Response(
+                OrderedDict(
+                    [
+                        ("count", cached_data.get("count")),
+                        ("next", cached_data.get("next")),
+                        ("previous", cached_data.get("previous")),
+                        ("results", cached_data.get("results")),
+                    ]
+                )
+            )
+
+        video = Video.objects.get(pk=self.request.user.id)
+
+        # we only want livesessions that are registered or with live_attendance not empty
+        queryset = LiveSession.objects.filter(
+            (
+                Q(is_registered=True)
+                | ~(Q(live_attendance__isnull=True) | Q(live_attendance__exact={}))
+            )
+            & Q(video__id=video.id)
+        )
+        page = self.paginate_queryset(self.filter_queryset(queryset))
+        serializer = serializers.LiveAttendanceGraphSerializer(
+            page,
+            many=True,
+            context={"video_timestamps": video.get_list_timestamps_attendences()},
+        )
+        # if the video is stopped, there is no need to limit the cache timeout
+        cache_timeout = (
+            None
+            if video.live_info and video.live_info.get("stopped_at")
+            else settings.VIDEO_ATTENDANCES_CACHE_DURATION
+        )
+        cache.set(
+            cache_key,
+            OrderedDict(
```

Comment:
you can put this in a variable and return this variable in a response object once cached. Otherwise you will compute twice for nothing at least `next` and `previous`